src/language_features/document_symbol.rs

```diff
@@ -265,7 +266,20 @@ pub fn format_symbol<T: Symbol<T>>(
                     .join("");
                 format!("{}{}", prefixing_hierarchy_symbols, last_hierarchy_symbol)
             };
-            let description = format!("{}{} ({:?})", hierarchy, symbol.name(), symbol.kind());
+
+            let mapped_symbol = symbols_mapping
+                .get(format!("{:?}", symbol.kind()).as_str())
+                .map_or(
+                    format!("{} ({:?})", symbol.name(), symbol.kind()),
+                    |mapped_kind| {
+                        if mapped_kind.is_empty() {
+                            symbol.name().to_string()
+                        } else {
+                            format!("{} {}", mapped_kind, symbol.name())
```

Comment:
A new case: prepend a user-string.

Reply:
> although some users may want to append the string, instead of prepending.

right.

I'm not sure we want to build a declarative config language for this,
because it will be limited.

We don't really need the full power of rust here;
we could do these text replacements in kakoune script.

I think we can achieve the same thing by overriding
`lsp-show-document-symbol`

Granted, that doesn't allow using different configurations multiple servers
in the same filetype like `lsp_servers` does.  But this issue is present
in a lot of our configuration switches.. multiple servers in LSP are hacky
and I don't expect much need for different per-server configuration here.
I think in future we should try to find a solution for this that doesn't
require stuffing everything into this weird `lsp_servers` TOML table in.

So I think we should try to find a way to customize `lsp-show-document-symbol`
per filetype (or per server) in a way that allows the user to run arbitrary
code.

perhaps something like (untested)

```kakrc
hook global BufCreate \*goto\* %{
	hook -once BufSetOption filetype=lsp-document-symbol %{
		hook -once NormalIdle .* %{
			execute-keys -draft %{%s \(Enum\)$<ret>d<a-b>ienum <esc>}
		}
	}
}
```

I didn't try this so I don't have a strong opinion yet.  I can see that
there are also upsides to the declarative approach: the UI boundary
is clearer and it's easier to not accidentally break it when changing
some internals.

Actually we should do this to get rid of one nested hook above

```diff
diff --git a/rc/lsp.kak b/rc/lsp.kak
index 5d38882be2..04236f8763 100644
--- a/rc/lsp.kak
+++ b/rc/lsp.kak
@@ -1225,7 +1225,7 @@
 }
 
 define-command -hidden lsp-show-document-symbol -params 3 -docstring "Render document symbols" %{
-    lsp-show-goto-buffer *goto* lsp-document-symbol %arg{1} %arg{3}
+    lsp-show-goto-buffer *symbols* lsp-document-symbol %arg{1} %arg{3}
     evaluate-commands -try-client %opt[toolsclient] %{
         set-option -add buffer path %arg{1} # for gf on the file name
         set-option buffer lsp_buffile %arg{2}
```

Reply:
I like this approach! I'll try to hack around the KakScript and see if it's possible to implement everything in terms of Kakoune UI only.

Reply:
One approach is to let the user (or default config) define a buffer local alias, but having to find the name to alias is extra burden. Matching against the buffer name is more obvious. Since the buffer is not a fifo buffer, it's not super obvious when the buffer will be populated (it won't be populated yet in `BufCreate`).
Maybe we should make it a fifo buffer for that reason, then we can use `BufCloseFifo`..
Either way, I think it's not tragic if it's not super obvious since the main way users learn about this is by copying the default Rust config.